log4j-api/src/main/java/org/apache/logging/log4j/util/PropertiesUtil.java

```diff
@@ -64,6 +68,8 @@ public final class PropertiesUtil {
     private static final Lazy<PropertiesUtil> COMPONENT_PROPERTIES =
             Lazy.lazy(() -> new PropertiesUtil(LOG4J_PROPERTIES_FILE_NAME, false));
 
+    private static final Pattern DURATION_PATTERN = Pattern.compile("([+-]?\\d+)\\s*(\\w+)?", Pattern.CASE_INSENSITIVE);
```

Comment:
*Nit:* I would rather dynamically build this pattern using the exact list of supported values from `TimeUnit`.

Reply:
I don't think such an optimization is necessary: the `getDurationProperty` method is only maintained for BC and is not used by any of our artifacts.

In Log4j Core I would prefer a standard `Duration.of` converter instead of this custom format.